rectools/models/sasrec.py

```diff
@@ -579,11 +822,13 @@ def __init__(
         n_heads: int,
         dropout_rate: float,
         use_pos_emb: bool = True,
+        use_ids_emb: bool = True,
+        use_cat_item_features_emb: bool = False,
         loss: str = "softmax",
         verbose: int = 0,
         cpu_n_threads: int = 0,
         transformer_layers_type: tp.Type[TransformerLayersBase] = SasRecTransformerLayers,  # SASRec authors net
-        item_net_type: tp.Type[ItemNetBase] = IdEmbeddingsItemNet,  # item embeddings on ids
+        item_net_type: tp.Type[ItemNetBase] = ConstructedItemNetBlock,  # item embeddings on ids and cat features
```

Comment:
My thoughts on interface (open for discussion):
How about we make instead one of these variants:
1. `item_net_constructor  = ItemNetConstructor(item_net_blocks=[IdEmbeddingsItemNet, CatFeaturesItemNet])`
2. `item_net_block_types = [IdEmbeddingsItemNet, CatFeaturesItemNet]`

`CatFeaturesItemNet` is what you've called `CatFeaturesEmebbedingsItemBlock`

`ItemNetConstructor` is what you've called `ConstructedItemNetBlock`. But the class should be universal and it doesn't have any `if` inside it's code. It just runs `from_dataset` methods of each block. And it summurizes all blocks `forward` results. Nothing else

Reply:
I did this now because I wanted to emphasis in the architecture of the net itself the names of the blocks that are used in the net. If it is not necessary, you can try one of the variants you suggested, but until it is hard to answer which one is better.